### PR TITLE
add nim books

### PR DESCRIPTION
**Pull request by me7**
## Hacktoberfest notes:

- due to volume of submissions, we may not be able to review PRs that do not pass tests and do not have informative titles.
- please read our [contributing guidelines](/CONTRIBUTING.md)
- be sure to check the output of Travis-CI for linter errors
- if this is your first open source contribution, make sure it's not your last!

## What does this PR do?
Add Resource(s)

## For resources
More book on Nim language

### Description 
This PR add more resource of Nim book, both online and PDF

### Why is this valuable (or not)?
New book have sample application created with Nim language, the old one is not

### How do we know it's really free?
Specify in resource

### For book lists, is it a book? For course lists, is it a course? etc.
Yes, it's book

### Checklist:
- [x] Not a duplicate
- [x] Included author(s) if appropriate
- [x] Lists are in alphabetical order
- [x] Needed indications added (PDF, access notes, under construction)

### CODE REVIEW

File: free-programming-books.md [hunk 1]
```diff
@@ -2025,7 +2025,9 @@ Kerridge (PDF) (email address *requested*, not required)
 
 ### Nim
 
-* [Nim Basics](https://narimiran.github.io/nim-basics/)
+* [Nim Basic (for beginner, PDF available)](https://narimiran.github.io/nim-basics/) - narimiran
```
Comment (eshellman) [inline free-programming-books.md:2028]: please don't invent titles

File: free-programming-books.md [hunk 2]
```diff
@@ -2025,7 +2025,9 @@ Kerridge (PDF) (email address *requested*, not required)
...
+* [Nim by Example](https://nim-by-example.github.io/) - Flaviu Tamas
```
Comment (eshellman) [inline free-programming-books.md:2029]: interesting that the linter doesn't flag the trailing slash here.

File: free-programming-books.md [hunk 3]
```diff
@@ -2025,7 +2025,9 @@ Kerridge (PDF) (email address *requested*, not required)
...
+* [Nim Days - Book teach to create several Nim app](https://xmonader.github.io/nimdays/) - Ahmed Thabet
```
Comment (eshellman) [inline free-programming-books.md:2030]: again, don't invent titles

**Review by eshellman — CHANGES_REQUESTED** [review]
Titles should be taken from the book itself - otherwise we can't identify duplicates.

**Comment by me7** [comment]
Thanks for review, I've change per your comment

**Comment by eshellman** [comment]
seems this failed because another merge broke the build.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**